test-app/ember-cli-build.js

```diff
@@ -13,8 +13,14 @@ module.exports = function (defaults) {
     // Add options here
   });
 
-  const { maybeEmbroider } = require('@embroider/test-setup');
-  return maybeEmbroider(app, {
+  const { Webpack } = require('@embroider/webpack');
+  return require('@embroider/compat').compatBuild(app, Webpack, {
```

Comment:
nay -- we still test ember-source 3.28 all the way to canary

Reply:
but only on embroider, which is not a very realistic test imo

Reply:
We don't have anything that the difference in build system needs to care about, so why make our test setup more complex?

in an addon like this, the build system is for _us_, the maintainers, to better debug and maintain the addon, not for declaring support.

Also, we get to lean on the fact that embroider-strict is the most correct thing we can test against right now (unless we completely switch over to vite), so _anything_ that works with embroider here is guaranteed to work with classic builds. It's the tests between embroider and ember-auto-import that guarantee that. We don't need to provide that coverage as well.

Reply:
if this was the builld system of the addon code itself I would agree but the purpose of this test-app is to simulate how the addon could be used in the wider ecosystem
you cannot say that every single app out there that might want to use tracked-buil-ins will be on fully static embroider which is what this change is saying

testing how classic build pipeline behaves with the output of the addon build step is the purpose of ember try - if I'm mistaken in any of the above statements please let me know, but otherwise I strongly believe this change is wrong

there is a reason why default v2 addon bp outputs a test app with maybeEmbroider and not fully static build code

Reply:
>  you cannot say that every single app out there that might want to use tracked-buil-ins will be on fully static embroider which is what this change is saying

naw, that's not what I'm saying -- I'm saying all embroider-strict compatible addons are compatible with less strict environments.

> there is a reason why default v2 addon bp outputs a test app with maybeEmbroider and not fully static build code

Because it hasn't been updated in ages, and is still _mostly_ the app blueprint from ember-cli.
There are _at least_ 4 things I change about every v2 addon I make, and this is one of them.
Even at work, I put all the test-apps on embroider-strict, and they work perfectly in the non-embroider consuming apps. 

But also, I haven't put much work in to wanting to update the blueprint myself, because i know we want to kill the test-app entirely, and have it be a vite environment in the same package as the addon itself -- removing the need for a monorepo entirely.

Now, vite _does_ support things that embroider-webpack and classic don't support yet, so unless you know what to avoid, it is actually dangerous right now to _only_ use vite (blindly).
Those features that embroider-vite has are:
- v2 addon w/ type=module support (this is broken in auto-import and embroider-webpack)
- I can't remember anything else rn, it might just be that